### PR TITLE
SDL updates

### PR DESCRIPTION
**Pull request by tpimh**
*This is just a draft for now, I will add more info below.*

List of SDL 1.2 packages:

- sdl
  - our version is 1.2.15, latest released tarball 1.2.15, [version in git](https://github.com/libsdl-org/SDL-1.2) is 1.2.16
  - building from [pspdev fork](https://github.com/pspdev/SDL) source code
  - license is `LGPL-2.1` (incorrectly indicated as `GPL2.1`)
- sdl_gfx
  - our version is 2.0.26, latest released tarball and [version in svn](https://sourceforge.net/p/sdlgfx/code/) is 2.0.27
  - building from [pspdev fork](https://github.com/pspdev/SDL_gfx/tree/SDL_gfx-psp) source code
  - license is `ZLIB`
- sdl_image
  - our version is 1.2.13, latest released tarball is 1.2.12, [version in git](https://github.com/libsdl-org/SDL_image/tree/SDL-1.2) is 1.2.13
  - building from upstream source code
  - license is `ZLIB` (indicated as `MIT`)
- sdl_mixer
  - our version is 1.2.14 (why?), latest released tarball is 1.2.12, [version in git](https://github.com/libsdl-org/SDL_mixer/tree/SDL-1.2) is 1.2.13
  - building from upstream source code
  - license is `ZLIB` (indicated as `custom`)
  - `wget` in prepare step
- sdl_net (added by this PR)
  - our version is 1.2.9, latest released tarball is 1.2.8, [version in git](https://github.com/libsdl-org/SDL_net/tree/SDL-1.2) is 1.2.9
  - building from upstream source code
  - license is `ZLIB`
- sdl_ttf
  - our version and the latest released tarball is 2.0.11, [version in git](https://github.com/libsdl-org/SDL_ttf/tree/SDL-1.2) is 2.0.11.1
  - building from a release from an archived [pspdev fork](https://github.com/pspdev/SDL_ttf)
  - license is `ZLIB` (indicated as `custom`)

List of SDL 2.0 packages:

- sdl2
  - our version is 2.30.2, latest release is 2.30.3
  - building from an upstream release tarball with patch
  - license is `ZLIB`
- sdl2_gfx
  - our version, latest released tarball and [version in svn](https://sourceforge.net/p/sdl2gfx/code/) is 1.0.4
  - building from [pspdev fork](https://github.com/pspdev/SDL_gfx) source code
  - license is `ZLIB`
- sdl2_image
  - our version is 2.6.3, latest release is 2.8.2
  - building from an upstream release tarball with patch
  - license is `ZLIB`
- sdl2_mixer
  - our version is 2.6.3, latest release is 2.8.0
  - building from an upstream release tarball
  - license is `ZLIB` (indicated as `MIT`)
- sdl2_net (added by this PR)
  - our version and latest release is 2.2.0
  - building from an upstream release tarball with patch
  - license is `ZLIB`
- sdl2_ttf
  - our version is 2.20.2, latest release is 2.22.0
  - building from an upstream release tarball with patch
  - license is `ZLIB` (indicated as `MIT`)

### CODE REVIEW

**Comment by sharkwouter** [comment]
With some of the license changes we have to be really careful here. If a library links to anything that has a different license, that can affect the licensing of the package. We should probably bundle the licenses of dependencies in the licenses directory of a package to be honest.

**Comment by tpimh** [comment]
I believe the licenses of these libraries were not changed, they were just indicated incorrectly. Anyway, Zlib is a very permissive license, and very similar to MIT, so this change shouldn't affect anything, and `GPL2.1` doesn't exist, it's just a typo.

File: sdl-mixer/PSPBUILD [hunk 1]
```diff
@@ -3,8 +3,8 @@ pkgver=1.2.14
 pkgrel=4
```
Comment (sharkwouter) [inline sdl-mixer/PSPBUILD:3]: pkgrel should be increased

**Review by sharkwouter — CHANGES_REQUESTED** [review]
There were still some pkgrel increases missing. Otherwise this looks good to me. I'll comment to explain the logic behind the weird licenses, but I'm okay with changing them with the PR.

**Comment by sharkwouter** [comment]
> I believe the licenses of these libraries were not changed, they were just indicated incorrectly. Anyway, Zlib is a very permissive license, and very similar to MIT, so this change shouldn't affect anything, and `GPL2.1` doesn't exist, it's just a typo.

What I meant was that because we can only build static binaries for the PSP, I set the license of the SDL packages to the one of the least permissive dependency, because it communicates to the developer using the library what rules they have to apply to. That might also be why I removed the L from LGPL, because when you statically linking, you are bound by the GPL when using LGPL licensed libraries.

I do think it makes sense to change this to the actual license of the library, but I don't know what other way we could use to communicate what license limitations your program will have to abide by when using a library.

**Comment by tpimh** [comment]
Yes, static linking is one of the pitfalls of GPL. But I believe there's a workaround for it: instead of linking statically, GPL modules can be built as PRXs and loaded dynamically (that requires some patching though). Also it makes sense to introduce flags (like in gentoo) to build packages with different optional dependencies.

**Comment by tpimh** [comment]
Also, the there's a problem with SDL 1.2 versions: they stopped adding new functions and creating new releases (version is frozen), but the code is still updated with fixes for new compilers and occasional backports from SDL2.

I can see several solutions there:

1. Use the last release version as `pkgver`, keep increasing `pkgrel` (problem with sdl_image and sdl_mixer as the version will go down)
2. Use the version indicated in changelog as `pkgver`, keep increasing `pkgrel` (problem only with sdl_mixer as the version will go down)
3. Use commit hash as `pkgver`, and start incrementing `epoch` every time it changes (version number doesn't make any sense anymore)

**Comment by tpimh** [comment]
I guess I should just skip checksum verification for all local files.

**Comment by sharkwouter** [comment]
I would just use the latest available version number for the package of SDL 1.2. That should be fine since they will not release new versions anyway.

**Comment by tpimh** [comment]
Makes sense, but then version numbers will decrement for two packages (possible problem for updates).

**Comment by sharkwouter** [comment]
> Makes sense, but then version numbers will decrement for two packages (possible problem for updates).

I think that's acceptable to do once.

**Comment by sharkwouter** [comment]
@tpimh could you still increase the pkgrel of the packages where that hasn't been done yet? Then we can get this merged, since it otherwise looks great. I just did a fix in SDL2 upstream which fixes SDL2main and I'd like to add that after merging this PR. Thanks in advance, I do really appreciate your work on this.

**Comment by tpimh** [comment]
I was thinking of actually updating every SDL package to the latest version, but I guess just bumping pkgrel is fine.

**Comment by sharkwouter** [comment]
Thanks! I appreciate the quick response. Updating can still be done in another PR.

**Pull request merged**